spec/lib/supplier_import_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+require './lib/supplier_import'
+require './lib/supplier_import_row'
+
+RSpec.describe SupplierImport do
+  describe '#initialize' do
+    it 'accepts CSV data as a string input' do
+      csv = "frameworkreference,suppliername\nRM1234,Cheese & Crackers LLP\nRM5678,Vaughan Legal\n"
+
+      expect { SupplierImport.new(csv) }.to_not raise_error
+    end
+  end
+
+  describe '#run!' do
+    it 'calls SupplierImportRow.import! for each row contained in the CSV' do
```

Comment:
this kinda describes the implementation. it might be more helpful as documentation and make it easier to change the implementation if it read as something like: `it 'creates supplier records for the suppliers listed in the CSV data`